packages/lib/gpcircuits/circuits/proto-pod-gpc.circom

```diff
@@ -125,11 +178,13 @@ template ProtoPODGPC (
             proofIndex <== entryProofIndex[entryIndex],
             proofSiblings <== entryProofSiblings[entryIndex]
         );
-
+    }
+    
+    for (var entryIndex = 0; entryIndex < MAX_ENTRIES + MAX_VIRTUAL_ENTRIES; entryIndex++) {
```

Comment:
I like the generality of this, though I have mixed feelings on locking ourselves into supporting all entry constraints on all types of virtual entries.  Let's go with it for now, but it's worth thinking about cost in future.  We may end up with some types of constraints only making sense to apply to real entries.  E.g. if we have lowerbound/upperbound constraints on numeric values, they wouldn't make sense to apply to public keys, content IDs, etc.  This might just mean we eventually end up with two different types of EntryConstraint module.

Relatedly, do you see virtual entries showing up in proof config with an object to hold constraints like equalsEntry?  That might imply supporting everything a user would expect to see there, though we'll want to rule some things out.  E.g. isOwnerID doesn't really make sense on the virtual entries we foresee.

Reply:
>I like the generality of this, though I have mixed feelings on locking ourselves into supporting all entry constraints on all types of virtual entries. Let's go with it for now, but it's worth thinking about cost in future. We may end up with some types of constraints only making sense to apply to real entries. E.g. if we have lowerbound/upperbound constraints on numeric values, they wouldn't make sense to apply to public keys, content IDs, etc. This might just mean we eventually end up with two different types of EntryConstraint module.

I agree, and unlike with entries we know the types beforehand, so it might be worth slimming this down later on down the line.

>Relatedly, do you see virtual entries showing up in proof config with an object to hold constraints like equalsEntry? That might imply supporting everything a user would expect to see there, though we'll want to rule some things out. E.g. isOwnerID doesn't really make sense on the virtual entries we foresee.

Yes. I was thinking it would be useful to support equality and list membership checks on signers' public keys.